### PR TITLE
Develop to Master (Updates)

### PR DESCRIPTION
**Pull request by mrz1836**
- Bumped crypto dep
- Added boiler plate go best practices
- Moved examples from readme into runnable files (easier to maintain long run)
- First version of updated readme (left some comments in the PR)
- Fix bug in test @jadwahab 
- Added dependabot config (manages new packages with PR reqs)
- Added basic share image for repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,100 +1,155 @@
 # go-bt
+> The go-to Bitcoin Transaction (BT) GoLang library  
+
+[![Release](https://img.shields.io/github/release-pre/libsv/go-bt.svg?logo=github&style=flat&v=1)](https://github.com/libsv/go-bt/releases)
+[![Build Status](https://travis-ci.com/libsv/go-bt.svg?branch=master&v=1)](https://travis-ci.com/libsv/go-bt)
+[![Report](https://goreportcard.com/badge/github.com/libsv/go-bt?style=flat&v=1)](https://goreportcard.com/report/github.com/libsv/go-bt)
+[![codecov](https://codecov.io/gh/libsv/go-bt/branch/master/graph/badge.svg?v=1)](https://codecov.io/gh/libsv/go-bt)
+[![Go](https://img.shields.io/github/go-mod/go-version/libsv/go-bt?v=1)](https://golang.org/)
+[![Sponsor](https://img.shields.io/badge/sponsor-libsv-181717.svg?logo=github&style=flat&v=3)](https://github.com/sponsors/libsv)
+[![Donate](https://img.shields.io/badge/donate-bitcoin-ff9900.svg?logo=bitcoin&style=flat&v=3)](https://gobitcoinsv.com/#sponsor)
```
Comment (mrz1836) [inline README.md:10]: Not sure if you want these last two badges... you need your org sponsorships activated etc

File: README.md [hunk 2]
```diff
@@ -1,100 +1,155 @@
...
-```console
-go get github.com/libsv/go-bt
-```
+<br/>
 
-## Tests
+### Features
+- Full Featured Bitcoin Transactions
+- Auto-Fee Calculations for Change Address
+- Uses common type: [`mAPI.Fee`](https://github.com/bitcoin-sv/merchantapi-reference/blob/35ba2d395acc632eed1bd528ab524aeafad8bd40/utils/types.go#L32) for easy integration across projects
```
Comment (mrz1836) [inline README.md:48]: @jadwahab please add more relevant features to mention

File: README.md [hunk 3]
```diff
@@ -1,100 +1,155 @@
...
+View the [examples](examples)
 
-	tx.AddOutput(o)
+<br/>
 
-	wif, _ := bsvutil.DecodeWIF("L3VJH2hcRGYYG6YrbWGmsxQC1zyYixA82YjgEyrEUWDs4ALgk8Vu")
+## Maintainers
+| [<img src="https://github.com/jadwahab.png" height="50" alt="JW" />](https://github.com/jadwahab) | [<img src="https://github.com/mrz1836.png" height="50" alt="MrZ" />](https://github.com/mrz1836) |
+|:---:|:---:|
+| [JW](https://github.com/jadwahab) | [MrZ](https://github.com/mrz1836) |
```
Comment (mrz1836) [inline README.md:138]: Anyone else to add here @jadwahab ?

File: README.md [hunk 4]
```diff
@@ -1,100 +1,155 @@
...
-```
+## Contributing
+View the [contributing guidelines](CONTRIBUTING.md) and please follow the [code of conduct](CODE_OF_CONDUCT.md).
 
-prints:
+### How can I help?
+All kinds of contributions are welcome :raised_hands:!
+The most basic way to show your support is to star :star2: the project, or to raise issues :speech_balloon:.
+You can also support this project by [becoming a sponsor on GitHub](https://github.com/sponsors/libsv) :clap:
+or by making a [**bitcoin donation**](https://gobitcoinsv.com/#sponsor) to ensure this journey continues indefinitely! :rocket:
```
Comment (mrz1836) [inline README.md:149]: Same link here, for where to donate lol

**Review by mrz1836 — COMMENTED** [review]
Some things to review @jadwahab

File: tx_test.go [hunk 5]
```diff
@@ -256,16 +452,17 @@ func TestTx_Change(t *testing.T) {
...
-			Fee Rate 0.8414 sat/B
-		*/
+		feePaid := tx.GetTotalInputSatoshis() - tx.GetTotalOutputSatoshis()
+		assert.Equal(t, uint64(122), feePaid)
+
+		txSize := len(tx.ToBytes())
+		assert.Equal(t, 251, txSize)
+
+		feeRate := float64(feePaid) / float64(txSize)
+		assert.Equal(t, 0.4860557768924303, feeRate)
```
Comment (mrz1836) [inline tx_test.go:466]: Am I wrong, this should be 0.50 or greater? A miner would not accept this tx?

**Review by mrz1836 — COMMENTED** [review]
Another question

File: tx_test.go (hunk 5 above)
Comment (mrz1836) [reply tx_test.go:466]: @jadwahab

File: README.md (hunk 4 above)
Comment (jadwahab) [reply README.md:153]: i'm fine to leave it on gobitcoinsv.com for now, enjoy haha

File: tx_test.go (hunk 5 above)
Comment (jadwahab) [reply tx_test.go:466]: it's got some data in there (which is priced at 0.25) which is why it's slightly less than 0.5

File: README.md (hunk 2 above)
Comment (jadwahab) [reply README.md]: 302e9e8

**Pull request merged**